staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -177,6 +177,32 @@ type Backoff struct {
 	Factor   float64       // Duration is multiplied by factor each iteration
 	Jitter   float64       // The amount of jitter applied each iteration
 	Steps    int           // Exit with error after this many steps
+	Cap      time.Duration // The maximum amount of time to wait if set
+}
+
+// Step returns the next interval in the exponential backoff.
+func (b *Backoff) Step() time.Duration {
+	if b.Steps < 1 {
+		if b.Jitter > 0 {
+			return Jitter(b.Duration, b.Jitter)
+		}
+		return b.Duration
+	}
+	b.Steps--
+	if b.Jitter > 0 {
+		b.Duration = Jitter(b.Duration, b.Jitter)
+	}
+	if b.Factor != 0 {
+		b.Duration = time.Duration(float64(b.Duration) * b.Factor)
```

Comment:
this means that the first sleep is `b.Factor * b.Duration`, not `b.Duration`, which is a change